### PR TITLE
Implement some new useful features

### PR DESCRIPTION
**Pull request by bsmithyman**
Includes some potential new features. Hold off on merging for now.

### CODE REVIEW

File: properties/base.py [hunk 1]
```diff
@@ -240,9 +241,11 @@ class Instance(basic.Property):
 
     info_text = 'an instance'
 
-    def __init__(self, help, instance_class, **kwargs):
+    def __init__(self, help, instance_class, serializer=None, **kwargs):
```
Comment (rowanc1) [inline properties/base.py]: this can just be a property and be dealt with in the kwargs

File: properties/base.py [hunk 2]
```diff
@@ -240,9 +241,11 @@ class Instance(basic.Property):
...
         assert isinstance(instance_class, type)
         self.instance_class = instance_class
+        if callable(serializer):
```
Comment (rowanc1) [inline properties/base.py]: this should be dealt with in the setter for `serializer`. I believe that the serializer should be named.

File: properties/base.py [hunk 3]
```diff
@@ -278,14 +281,20 @@ def assert_valid(self, instance, value=None):
             value.validate()
         return True
 
-    @staticmethod
-    def as_json(value):
+    def as_json(self, value):
```
Comment (rowanc1) [inline properties/base.py]: currently there is a pickle and a serialize call that are different but really doing the same thing.

File: properties/utils.py [hunk 4]
```diff
@@ -43,6 +43,12 @@ def func_wrapper(self):
 
     return func_wrapper
 
+def filter_dict(cls, fields):
```
Comment (rowanc1) [inline properties/utils.py]: `filter_props`?

Comment (bsmithyman) [inline properties/utils.py]: Could do? I thought that `filter_dict` would be a bit clearer (that it is working on the serialized version). Not really that concerned one way or another about the name, though. Could also make it something like `filter_keys`.

Comment (fwkoch) [inline properties/utils.py]: I co-opted this function into the 0.2.3 release: https://github.com/3ptscience/properties/blob/master/properties/utils.py#L54

File: properties/base.py (hunk 3 above)
Comment (bsmithyman) [inline properties/base.py]: Can you clarify what you want here? It's not clear to me that we necessarily want to be overriding the pickle behaviour if we are setting a JSON serializer...?

**Comment by fwkoch** [comment]
This has been made obsolete by #63 (tasks) and #64 (serialization).

**Pull request closed without merging**